src/Umbraco.Web/WebBootManager.cs

```diff
@@ -533,12 +525,42 @@ protected override void InitializeResolvers()
                 new DefaultCultureDictionaryFactory());
         }
 
+        /// <summary>
+        /// The method used to create indexes on a cold boot
+        /// </summary>
+        /// <remarks>
+        /// A cold boot is when the server determines it will not (or cannot) process instructions in the cache table and
+        /// will rebuild it's own caches itself.
+        /// </remarks>
+        public static IEnumerable<BaseIndexProvider> GetIndexesForColdBoot()
+        {
+            // NOTE: This is IMPORTANT! ... we don't want to rebuild any index that is already flagged to be re-indexed
+            // on startup based on our _indexesToRebuild variable and how Examine auto-rebuilds when indexes are empty.
+            // This callback is used above for the DatabaseServerMessenger startup options.
+
+            // all indexes
+            IEnumerable<BaseIndexProvider> indexes = ExamineManager.Instance.IndexProviderCollection;
+
+            // except those that are already flagged
+            // and are processed in Complete()
+            if (IndexesToRebuild.Any())
+                indexes = indexes.Except(IndexesToRebuild);
+
+            // return
+            foreach (var index in indexes)
+                yield return index;
+
+            // and clear
+            IndexesToRebuild.Clear();

```

Comment:
Yeah, because I had to make it static so that WebBootManager.GetIndexesForColdBoot can be used by a dev's app startup. We don't really expose the current IBootManager anywhere apart from the global.asax Application class and I thought that might be too annoying to use for devs instead of just using a normal Application event handler.